esmvalcore/preprocessor/_time.py

```diff
@@ -552,25 +553,32 @@ def climate_statistics(cube,
             time_weights = get_time_weights(cube)
             if time_weights.min() == time_weights.max():
                 # No weighting needed.
-                cube = cube.collapsed('time', operator_method)
+                clim_cube = cube.collapsed('time', operator_method)
             else:
-                cube = cube.collapsed('time',
-                                      operator_method,
-                                      weights=time_weights)
+                clim_cube = cube.collapsed('time',
+                                           operator_method,
+                                           weights=time_weights)
         else:
-            cube = cube.collapsed('time', operator_method)
-        return cube
-
-    clim_coord = _get_period_coord(cube, period, seasons)
-    operator = get_iris_analysis_operation(operator)
-    clim_cube = cube.aggregated_by(clim_coord, operator)
-    clim_cube.remove_coord('time')
-    if clim_cube.coord(clim_coord.name()).is_monotonic():
-        iris.util.promote_aux_coord_to_dim_coord(clim_cube, clim_coord.name())
+            clim_cube = cube.collapsed('time', operator_method)
     else:
-        clim_cube = iris.cube.CubeList(clim_cube.slices_over(
-            clim_coord.name())).merge_cube()
-    cube.remove_coord(clim_coord)
+        clim_coord = _get_period_coord(cube, period, seasons)
+        operator = get_iris_analysis_operation(operator)
+        clim_cube = cube.aggregated_by(clim_coord, operator)
+        clim_cube.remove_coord('time')
+        if clim_cube.coord(clim_coord.name()).is_monotonic():
+            iris.util.promote_aux_coord_to_dim_coord(clim_cube,
+                                                     clim_coord.name())
+        else:
+            clim_cube = iris.cube.CubeList(
+                clim_cube.slices_over(clim_coord.name())).merge_cube()
+        cube.remove_coord(clim_coord)
```

Comment:
This line looks a bit fishy to me, but it was there before, so there is that. It looks like this should be `clim_cube.remove_coord(clim_coord)`. @valeriupredoi, this seems to come from you. Do you have an opinion?